src/serialize.h

```diff
@@ -571,7 +571,7 @@ void Serialize_impl(Stream& os, const prevector<N, T>& v, const unsigned char&)
 {
     WriteCompactSize(os, v.size());
     if (!v.empty())
-        os.write((char*)&v[0], v.size() * sizeof(T));
+        os.write((char*) v.data(), v.size() * sizeof(T));
```

Comment:
I wonder why we're casting to `(char*)` here iso `(const char*)`, even though these are write operations and the type passed in is `const`.

There's a couple more of these. Though seems out of scope to fix in this pull.